dlib/dnn/loss.h

```diff
@@ -389,7 +389,8 @@ namespace dlib
     {
     public:
 
-        typedef weighted_label<unsigned long> training_label_type;
+        typedef dlib::weighted_label<unsigned long> weighted_label;
```

Comment:
I added this typedef, because maybe users would expect this loss to behave as the `loss_multiclass_log_per_pixel_weighted_`.
Sorry for doing this in this PR...

Reply:
Na, seems good.  Update the _abstract.h docs too though.